instrumentation/graphql/lib/opentelemetry/instrumentation/graphql/tracers/graphql_trace.rb

```diff
@@ -36,12 +36,13 @@ def parse(query_string:, &block)
           def validate(query:, validate:, &block)
             tracer.in_span('graphql.validate') do |span|
               super.tap do |response|
-                errors = response[:errors]&.compact&.map(&:to_h)&.to_json
-                unless errors.nil?
+                errors = response[:errors]&.compact&.map(&:to_h) || []
+
+                unless errors.empty?
                   span.add_event(
                     'graphql.validation.error',
                     attributes: {
-                      'message' => errors
+                      'message' => errors.to_json
```

Comment:
Yep: https://github.com/rmosolgo/graphql-ruby/blob/bde9b64d7df0cdc28045a9231d3219a800052ffc/lib/graphql.rb#L3

Reply:
I hate to do this but I am feeling a little worried about JSON serialization failing for some reason. Is there anything in the `errors` object that may link back to objects generated by frameworks like Rails? 

I have been bitten too many times by `as_json called on X without :only option. You must explicitly declare the keys you wish to serialize`

Reply:
Something else comes to mind. Is there a community consensus around semconv for these kinds of validation errors? 
Should we submit an otep for this? cc: @open-telemetry/ruby-contrib-approvers ?

Reply:
> I hate to do this but I am feeling a little worried about JSON serialization failing for some reason. Is there anything in the errors object that may link back to objects generated by frameworks like Rails?

I don't think that's possible since this is only for static validation errors specifically, which are built by the graphql-ruby gem. Here's the representation: https://github.com/rmosolgo/graphql-ruby/blob/f05ea569af0ceea6eea41919b7bb61f3eb06f2cd/lib/graphql/static_validation/error.rb

Example of one: https://github.com/rmosolgo/graphql-ruby/blob/f05ea569af0ceea6eea41919b7bb61f3eb06f2cd/lib/graphql/static_validation/rules/argument_literals_are_compatible_error.rb#L21-L41

This `to_json` existed before as well so it's not new.

Reply:
OK thanks. I have created a discussion here in the hopes to get some eyes/opinions about what to do about this going forward. 

https://github.com/open-telemetry/opentelemetry-specification/discussions/3624

Reply:
Seems like there is a tracking issue for this now: https://github.com/open-telemetry/semantic-conventions/issues/182

Given that there isn't a consensus and the semconv is evolving I am going to take another look at the PR Thank you again for your contributions!

Reply:
This is rather curious: 

https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/plugins/node/opentelemetry-instrumentation-graphql/src/instrumentation.ts#L393

It records the validation errors using `recordException` without a backtrace.